usaspending_api/etl/management/commands/load_submission.py

```diff
@@ -755,6 +445,383 @@ def dictfetchall(cursor):
+
+        # Create the place of performance location
+        pop_location, created = get_or_create_location(
+            place_of_performance_field_map, row, place_of_performance_value_map)
+
+        # Find the award that this award transaction belongs to. If it doesn't exist, create it.
+        awarding_agency = Agency.objects \
+            .filter(
+                toptier_agency__cgac_code=row['awarding_agency_code'],
+                subtier_agency__subtier_code=row["awarding_sub_tier_agency_c"]) \
+            .first()
+        created, award = Award.get_or_create_summary_award(
+            awarding_agency=awarding_agency,
+            piid=row.get('piid'),
+            fain=row.get('fain'),
+            uri=row.get('uri'),
+            parent_award_id=row.get('parent_award_id'))
+        award.save()
+
+        AWARD_UPDATE_ID_LIST.append(award.id)
+        AWARD_CONTRACT_UPDATE_ID_LIST.append(award.id)
+
+        parent_txn_value_map = {
+            "award": award,
+            "awarding_agency": awarding_agency,
+            "funding_agency": Agency.objects.filter(toptier_agency__cgac_code=row['funding_agency_code'],
+                                                    subtier_agency__subtier_code=row["funding_sub_tier_agency_co"]).first(),
+            "recipient": legal_entity,
+            "place_of_performance": pop_location,
+            'submission': submission_attributes,
+            "period_of_performance_start_date": format_date(row['period_of_performance_star']),
+            "period_of_performance_current_end_date": format_date(row['period_of_performance_curr']),
+            "action_date": format_date(row['action_date']),
+        }
+
+        transaction_instance = load_data_into_model(
+            Transaction(), row,
+            field_map=contract_field_map,
+            value_map=parent_txn_value_map,
+            as_dict=True)
+
+        transaction_instance, created = Transaction.objects.get_or_create(**transaction_instance)
+
+        contract_value_map = {
+            'transaction': transaction_instance,
+            'submission': submission_attributes,
+            'reporting_period_start': submission_attributes.reporting_period_start,
+            'reporting_period_end': submission_attributes.reporting_period_end,
+            "period_of_performance_potential_end_date": format_date(row['period_of_perf_potential_e'])
+        }
+
+        contract_instance = load_data_into_model(
+            TransactionContract(), row,
+            field_map=contract_field_map,
+            value_map=contract_value_map,
+            save=True)
+
+    # Update awards for new linkages
+    update_awards(tuple(AWARD_UPDATE_ID_LIST))
+    update_contract_awards(tuple(AWARD_CONTRACT_UPDATE_ID_LIST))
+
+    # Update the descriptions TODO: If this is slow, add ID limiting as above
+    update_model_description_fields()
```

Comment:
Should this be in load_d1? Should we run this after all records have been created (i.e. after all files have been loaded)